### PR TITLE
Improvements to editor JSON handling

### PR DESCRIPTION
**Pull request by mwiencek**
See commits, but in summary:
 * Makes `Editor::TO_JSON` output a sanitized editor object by default; you have to serialize a full one with `$c-> unsanitized_editor_json`. And this method only fills in the `email` and `birth_date` if the active user is the same as the serialized one, or is an account admin (in the case of email).
 * In development and testing mode, throws an error if any editor objects inside hydration props contain unsanitized data.

### CODE REVIEW

File: root/types.js [hunk 1]
```diff
@@ -998,7 +998,7 @@ declare type SanitizedCatalystContextT = {
     +current_language: string,
     +genre_map?: {+[genreName: string]: GenreT, ...},
   },
-  +user: ActiveSanitizedEditorT | null,
+  +user: ActiveEditorT | null,
 };
 
 declare type SanitizedEditorPreferencesT = {
```
Comment (reosarevok) [inline root/types.js:1004]: Shouldn't this be renamed too?

**Review by reosarevok — COMMENTED** [review]
Did you check and change every place where we used unsanitized data on the server to see if they can be made sanitized? I saw you changed some, just not sure how thorough you were with it :)

**Comment by mwiencek** [comment]
> Did you check and change every place where we used unsanitized data on the server to see if they can be made sanitized? I saw you changed some, just not sure how thorough you were with it :)

I believe I did, because I started by globally changing every `EditorT` type to `SanitizedEditorT`, and then only reverting the ones that errored due to missing information.

However, we can further improve the unsanitized cases by passing only the specific data needed (usually just preferences or privilege flags) as a separate prop.

**Comment by jesus2099** [comment]
This PR is not merged yet I see a related error on https://musicbrainz.org/account/applications :

```
Internal Server Error

Can't call method "sanitized_json" on an undefined value at lib/MusicBrainz/Server/Entity/EditorOAuthToken.pm line 120. at lib/MusicBrainz/Server/Entity/EditorOAuthToken.pm line 120
JSON::__ANON__(?, ?) called at lib/MusicBrainz/Server/Entity/Util/JSON.pm line 37
MusicBrainz::Server::Entity::Util::JSON::encode_with_linked_entities(?, ?) called at lib/MusicBrainz/Server/Renderer.pm line 25
MusicBrainz::Server::Renderer::send_to_renderer(?, ?, ?) called at lib/MusicBrainz/Server/Renderer.pm line 58
MusicBrainz::Server::Renderer::render_component(?, ?, ?) called at lib/MusicBrainz/Server/View/Node.pm line 20
Catalyst::dispatch(?) called at lib/MusicBrainz/Server.pm line 325
MusicBrainz::Server::__ANON__ at lib/MusicBrainz/Server.pm line 300
MusicBrainz::Server::with_translations(?, ?) called at lib/MusicBrainz/Server.pm line 326
Class::MOP::Method::Wrapped::__ANON__(?) called at lib/MusicBrainz/Server.pm line 340
Class::MOP::Method::Wrapped::__ANON__(?) called at lib/MusicBrainz/Server.pm line 418
```

Same on beta.

**Review by yvanzo — APPROVED** [review]

**Pull request merged**